pkg/controller/deliverable/reconciler.go

```diff
@@ -260,6 +265,40 @@ func (r *Reconciler) trackDependencies(deliverable *v1alpha1.Deliverable, realiz
 	}
 }
 
+func (r *Reconciler) cleanupOrphanedObjects(ctx context.Context, previousResources, realizedResources []v1alpha1.RealizedResource) error {
+	log := logr.FromContextOrDiscard(ctx)
+
+	var orphanedObjs []*corev1.ObjectReference
+	for _, prevResource := range previousResources {
+		orphaned := true
+		for _, realizedResource := range realizedResources {
+			if realizedResource.StampedRef.Kind == prevResource.StampedRef.Kind &&
+				realizedResource.StampedRef.Name == prevResource.StampedRef.Name {
```

Comment:
wdyt of considering the whole `apiVersion` here? e.g., to account for version bumps (like how we bumped `kpack/v1alpha1` to `kpack/v1alpha2` a couple months ago), or even group changes